### PR TITLE
add race model tests

### PR DESCRIPTION
**Pull request by willblake01**
Provides better test coverage for the race model.

-Add traits to race factory for testing
-Add tests to race_spec

### CODE REVIEW

**Review by runbeyond — APPROVED** [review]

File: app/models/race.rb [hunk 1]
```diff
@@ -32,7 +32,7 @@ def name_with_event
   end
 
   def generate_slug
-    "#{starts_at.year} #{event.name} #{name}".parameterize
+    "#{starts_at.year} #{event.name} #{name}".parameterize if @race
```
Comment (ColinW520) [inline app/models/race.rb:35]: `@race` will always be `nil` here... not sure how this would ever have worked?

Comment (willblake01) [reply app/models/race.rb:35]: I get this error without it:
image

Comment (ColinW520) [reply app/models/race.rb:35]: Sounds like something that needs to be resolved with setting data in the factory. Really don't want to be using @vars in models.

Comment (willblake01) [reply app/models/race.rb]: Thanks! I felt like that was a little bit hacky but I figured out a much better solution. I added a safe navigator operator to the generate_slug method and it worked like a charm.

File: app/models/race.rb [hunk 2]
```diff
@@ -32,7 +32,7 @@ def name_with_event
   end
 
   def generate_slug
-    "#{starts_at.year} #{event.name} #{name}".parameterize
+    "#{starts_at&.year} #{event&.name} #{name}".parameterize
```
Comment (ColinW520) [inline app/models/race.rb:35]: Why the safe-nav here? If anything, those values should _always_ be present before slugging.

Comment (willblake01) [reply app/models/race.rb:35]: I was getting that nil class error above. I replaced the `if @race` line in the model with the safe-nav operators. Both resolved the error for the test but I'll keep looking at it. I guess I don't have my mind fully wrapped around what's going on with the error.

**Pull request closed without merging**